packages/react-components/react-tree/src/utils/createFlatTreeItems.ts

```diff
@@ -38,8 +38,9 @@ export function createFlatTreeItems<Props extends FlatTreeItemProps<unknown>>(
       }
       break;
     }
-    const isLeaf =
-      treeItemProps.leaf ?? (treeItemProps.value === undefined || nextItemProps?.parentValue !== treeItemProps.value);
+    const itemType =
+      treeItemProps.itemType ??
```

Comment:
since we building the tree directly from datastructure, should we let users override itemType? it kinda feels like we should just validate that their input is correct and generate the itemType correctly.

The data is already there to make the decision in the case of flat tree